File: backend/src/main/java/com/pranay/ems/service/EmployeeService.java
package com.pranay.ems.service;

import com.pranay.ems.dto.request.EmployeeRequest;
import com.pranay.ems.dto.response.EmployeeResponse;
import com.pranay.ems.enums.EmployeeStatus;
import org.springframework.data.domain.Page;

import java.util.List;

public interface EmployeeService {

    EmployeeResponse addEmployee(EmployeeRequest request);

    List<EmployeeResponse> getAllEmployees();

    EmployeeResponse getEmployeeById(Long id);

    EmployeeResponse updateEmployee(Long id, EmployeeRequest request);

    void deleteEmployee(Long id);

    List<EmployeeResponse> searchEmployees(String keyword);
    Page<EmployeeResponse> getEmployees(
            int page,
            int size,
            String sortBy,
            String direction
    );
    List<EmployeeResponse> getEmployeesByStatus(EmployeeStatus status);
    List<EmployeeResponse> getEmployeesByDesignation(String designation);
}

File: frontend/src/services/employeeService.js
import api from "../api/axios";

/* ==============================
   Get All Employees
================================ */

export const getAllEmployees = async () => {

    const response = await api.get("/api/v1/employees");

    return response.data;

};

/* ==============================
   Get Employee By Id
================================ */

export const getEmployeeById = async (id) => {

    const response = await api.get(`/api/v1/employees/${id}`);

    return response.data;

};

/* ==============================
   Create Employee
================================ */

export const createEmployee = async (employee) => {

    const response = await api.post(
        "/api/v1/employees",
        employee
    );

    return response.data;

};

/* ==============================
   Update Employee
================================ */

export const updateEmployee = async (id, employee) => {

    const response = await api.put(
        `/api/v1/employees/${id}`,
        employee
    );

    return response.data;

};

/* ==============================
   Delete Employee
================================ */

export const deleteEmployee = async (id) => {

    const response = await api.delete(
        `/api/v1/employees/${id}`
    );

    return response.data;

};

/* ==============================
   Search Employees
================================ */

export const searchEmployees = async (keyword) => {

    const response = await api.get(
        `/api/v1/employees/search?keyword=${keyword}`
    );

    return response.data;

};